### ISSUE TITLE
Brave browser new tab not launching into Google search box

### ISSUE THREAD

**Issue by davidwf1** [brave/brave-browser#17358]
## Description 
Brave browser new tab opens the required website (Google.co.uk) but remains in address bar.

## Steps to Reproduce

   1. In brave://settings/appearance set https://google.co.uk as home button
   2. In brave://settings/newTab set New page shows setting to homepage
   3. open new tab with "+"
   4. result: cursor remains in address bar rather than into Google search box

## Actual result:

![image](https://user-images.githubusercontent.com/88488205/128336788-6150aaed-6141-4c84-80ed-c1d2c0375b11.png)

## Expected result:
![image](https://user-images.githubusercontent.com/88488205/128336874-297b941d-9fe9-4177-894a-f02a18872c12.png)

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by rebron** [comment]
This is a focus issue. When a user enables the home button and sets their homepage to `google.com` or `google.co.uk`, when selecting the home button, the `focus is in the search box on the page`.

If user sets their new tab page to be the same as the homepage, `google.com` or `google.co.uk` in this case, `opening a new tab sets focus in the url bar.`
 
Request is to match the focus behavior like when selecting the home button.

<img width="712" alt="Screen Shot 2021-08-06 at 10 13 35 AM" src="https://user-images.githubusercontent.com/5951041/128547825-84b0b7c0-60aa-49d9-ab0c-3c6a3270c191.png">

**Comment by JimBreton** [comment]
I can understand the desire for the requested behavior, but are there any studies possible that would indicate user impact? Personally, when I open a 'New Tab' my immediate next action is to start typing in the URL bar -- whether it be to use the built-in search behavior, or to begin typing a URL.  I believe implementing this request would break that.  Not sure what 'typical' user behavior patterns look like.  I suppose for someone who goes to the effort of changing the NTP it might make more sense to use the focus hint provided by the custom page, so there is that.

**Comment by davidwf1** [comment]
Yes, I agree with those findings……clicking the home button does (in my case) result in the cursor being in the Google search box

Are you saying that is the correct function?

**Comment by dhulihan** [comment]
I've also run into this, and it's a bit annoying when using a non-search engine landing page for your homepage (for example, when using custom startpage with an input field).

**Comment by sid760** [comment]
I also wanted the same feature on brave, and found the following solution:
Just press `Esc` key after opening a new tab and the focus will be set to the google search bar
Working: 
1. Opened a new tab (focus was on address bar)
![image](https://user-images.githubusercontent.com/43800138/202846481-fa5c3131-e88f-41ce-bbd4-c6c8fe85cc9c.png)

2. Pressed `esc` key
![image](https://user-images.githubusercontent.com/43800138/202846526-1084daba-ea0b-4e74-ab8a-3f21f5485ddb.png)
and the focus was set to google search box

Works for me (it's better not having to touch mouse), altough it would be better not having to press `esc` key even, just like chrome.

**Comment by davidwf1** [comment]
nope, not for me....I get a blank screen

<img width="1049" alt="image" src="https://user-images.githubusercontent.com/88488205/202847104-9257f2d6-83ed-4026-9c4d-f539078044c7.png">

**Comment by sid760** [comment]
> nope, not for me....I get a blank screen

@davidwf1  
Well you have to set the google as home screen first and set home screen as new tab,

Here are the steps: 
1. Open `brave://settings/appearance` via the address bar.
2. Enable `Show home button` .
3. Choose Enter custom web address → enter `www.google.com`
4. Open `brave://settings/newTab` via the address bar.

New tab page shows → Homepage

**Comment by davidwf1** [comment]
ah....yes, yes, yes....thank you :)
I had the home button enabled but had not set the page to Google!....

**Comment by davidwf1** [comment]
Can that also be set on Android?.....have looked but cant seem to find it

**Comment by LevYas** [comment]
This is still relevant. I use a custom start page with keyboard commands so I want to have default focus on the start page and not on the address bar.
It's fine if that will work via an extra flag or setting.

**Comment by salwinat0r** [comment]
> I use a custom start page with keyboard commands so I want to have default focus on the start page and not on the address bar.

Would you mind sharing how you achieved this?

**Comment by LevYas** [comment]
> Would you mind sharing how you achieved this?

Achieved what? I did not make the focus issue work like I wanted, so I stopped using the custom start page. Browser remembers recent URLs so it's similar to typing the first letter or two in the custom start page.

**Comment by IMNdi** [comment]
When I press new tab, the page gets the focus. This is annoying to me as my home page land has no search.

Why no setting? At least a flag.

Seeing how people complained about this in 2021, perhaps adding a search bar to my home page would be prudent.

**Comment by e-vishnu-gh** [comment]
Oh! let me be the comment for year 2025! 

🤞 for 2026

**Comment by rebron** [comment]
cc: @zenparsing only via flag option

**Comment by zenparsing** [comment]
@rebron 

Normally, when the user opens a new tab focus is given to the omnibox. In order to circumvent this behavior, we could make the following patch to `Browser::ShouldFocusLocationBarByDefault`:

```cpp
bool Browser::ShouldFocusLocationBarByDefault(WebContents* source) {
// ...
    if ((url.SchemeIs(content::kChromeUIScheme) &&
         url.host() == chrome::kChromeUINewTabHost) ||
        (virtual_url.SchemeIs(content::kChromeUIScheme) &&
         virtual_url.host() == chrome::kChromeUINewTabHost)) {
      // Add a Brave-specific patch here that does the following test:
      if (auto ntp_url = brave::GetNewTabPageURL(profile());
          ntp_url.is_valid() && ntp_url != url && ntp_url != virtual_url) {
        return false;
      }
      return true;
    }
// ...
}
```

Note that this feature may be problematic for a couple of reasons:

- In order to enable this behavior, you have to temporarily show the "Home" button on the appearance page and set the home URL. You also have to set the "New tab page shows" option on the "Get started" page.
- When a custom homepage is set for the NTP, there is a noticeable delay between when the tab is created and when the content is loaded. During this time the focus is nowhere.
- If the custom homepage does not have a default focus element, focus will be just be "on the page", and not a particular element.

**Comment by DecapitatedKneecap** [comment]
how is this still an issue 5 1/2 years later?

whenever i open my google bookmark from the bookmark bar in a new tab and then click the tab, the google search box is not active. oddly, i can see the cursor in the search box for a split second before it disappears as the focus is transferred elsewhere. this means my typing goes into the void before i realize the search box isn't even active.

the focus seems to actually end up on the icon to the left of the address bar. typing or arrow keys do nothing:

<img width="148" height="45" alt="Image" src="https://github.com/user-attachments/assets/40c56cbb-88dc-413a-9b69-d5caceba12e1" />

however, i can hit ctrl+space to open this dropdown list:

<img width="357" height="351" alt="Image" src="https://github.com/user-attachments/assets/44f3b408-b6bd-469c-ae02-824658a8685b" />

it only seems to do this once per instance. all consecutive new tab openings of my google bookmark will activate the google search box as expected. this instance phenomenon also affects private windows where opening google in a private window will initially exhibit the behavior. subsequent instances of private window tabs do not. this is also per profile

i wish this would get fixed. it is highly annoying to type out my entire query only for it to go nowhere. it's 2026. let's get this fixed

**Comment by DecapitatedKneecap** [comment]
not a single response in over 30 days and it's still doing it
